src/Lykke.Service.Dash.Api/Services/DashService.cs

```diff
@@ -113,27 +113,22 @@ public Transaction GetTransaction(string transactionHex)
 
         public async Task BroadcastAsync(Transaction transaction, Guid operationId)
         {
-            TxBroadcast response;
+            TxBroadcast response = null;
 
             try
             {
                 response = await _dashInsightClient.BroadcastTxAsync(transaction.ToHex());
-
-                if (response == null)
-                {
-                    throw new ArgumentException($"{nameof(response)} can not be null");
-                }
-                if (string.IsNullOrEmpty(response.Txid))
-                {
-                    throw new ArgumentException($"{nameof(response)}{nameof(response.Txid)} can not be null or empty. Response={response}");
-                }
             }
             catch (Exception ex)
             {
-                await _log.WriteErrorAsync(nameof(DashService), nameof(BroadcastAsync),
-                    $"transaction: {transaction}, operationId: {operationId}", ex);
-
-                throw;
+                if (ex.ToString().Contains("transaction already in block chain"))
```

Comment:
Looks brittle. There is no more specific exception with something kinda error code?

Reply:
Yeah. Agree. Node returns 400 error and the body contains this description... It is like a quick fix for test. I am planning to add normal check if hash is already in blockchain or not before brodcast